server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -130,6 +131,12 @@ public class RemoteClusterPortSettings {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<ByteSizeValue> MAX_HEADER_SIZE = Setting.byteSizeSetting(
+        REMOTE_CLUSTER_PREFIX + "max_header_size",
+        new ByteSizeValue(16, ByteSizeUnit.KB),
```

Comment:
16KB is probably a bit too limiting for transport headers. I think we can go for something like 1MB. For RCS 2.0, we send the entire serialized authentication which could a bit verbose. For example, if it is an API key created by an SAML user with a complex set of role descriptors (e.g. a long list of FLS/DLS), the authentication object will contain all user metadata, both role descriptors and limitBy role descriptor bytes and potentially some more metadata for the API key itself. In addition, we also send a role descriptor list for remote indices privileges which could be sizable as well if the privileges are complex enough.

Reply:
Thank you for the explanation of the header contents for RCS 2.0 connections.

I'll go on with your suggestion, but I will also reach out to see what feedback we get from folks that look at it from a network attack perspective.

Reply:
@ywangd Is it worth looking into a small example for such headers and try to extrapolate from that? That may also help on the sender side where we could log a warning message if headers are above a certain threshold (in relation to the default value of this setting).

In the mean time I'll go with 1MB (1000s of connections required for DOS).

Reply:
Yang ran some great investigations to obtain some typical header sizes, and we eventually settled on 64KB.